services/event_service.py

```diff
@@ -47,3 +47,18 @@ def delete_event(event_id: int, db: Session):
 
     db.delete(event)
     db.commit()
+
+def update_event(event_id:int,data:EventUpdate,db:Session):
+    event = db.query(Event_DB).filter_by(id=event_id).one_or_none()
+    if event is None:
+        raise HTTPException(status.HTTP_404_NOT_FOUND)
+    if len(data.description_en) != 0:

```

Comment:
And with optional fields on the Update schema: Replace 0 length checks with Check if None. The client will send only what they intend to change.